### PR TITLE
83 create db table for contracts

### LINKED ISSUE #83: Create db table for contracts
## Description
As an admin I want my contracts to persist between restarts of the server so that I don't have to recreate the contracts each time.

## Acceptance criteria
- [x] The server can be restarted and the contract data (the fields in the contract class) are persisted
- [x] All fake repository methods should use the actual database (except recents and favorites)

## Dependencies
- #84

### PR DESCRIPTION
**Pull request by martinjonsson01**
## Summary
Refactors the Contract entity to contain a list of Tag entities instead of a list of strings.
Refactors the ListInput-component (the one used for inputting tags) to support conversion from and to any type. 
Adds a connection to the PostgreSQL database for Contracts.

Currently the repository tests run on the same local database that is used for running the web server, but to minimize the size of this PR I'm going to fix that in a hotfix-branch.

## Definition of done
- [x] All acceptance criteria are completed.
- [x] The code has (at least) 80% branch test coverage.
- [x] All tests pass.
- [ ] The code is reviewed and accepted by (at least) two other team members.
- [x] Views adhere to design style.

Closes #83

### CODE REVIEW

**Review by jonathan-lindqvist — APPROVED** [review]
lgtm

File: src/Infrastructure/Contracts/PostgresContractRepository.cs [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+    /// <inheritdoc />
+    public void AddRecent(Contract contract)
+    {
+        _recent.Add(contract);
+    }
+
+    /// <inheritdoc />
+    public bool Remove(Guid id)
+    {
+        _recent.Remove(id); // This line should not exist when there is an actual database (it will remove any relations)
```
Comment (martinjonsson01) [inline src/Infrastructure/Contracts/PostgresContractRepository.cs:62]: nit:
```suggestion
        _recent.Remove(id); // This line should not exist when recents are included in the database
```

**Review by JacobBredin — APPROVED** [review]
LGTM! Small question: Why tags? or whats the problem that's being solved?

File: src/Domain/Contracts/Tag.cs [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+{
+    /// <summary>
+    /// Gets or sets the identifier.
+    /// </summary>
+    public Guid Id { get; set; } = Guid.NewGuid();
+
+    /// <summary>
+    /// Gets or sets the identifier of the <see cref="Contract"/> which this tag belongs to.
+    /// </summary>
+    public Guid ContractId { get; set; } = Guid.NewGuid();
```
[the remark this answers is not in the document]
Comment (martinjonsson01) [reply src/Domain/Contracts/Tag.cs:16]: Then they'll have different tag entities with the same text content. Right now we've got no reason to make sure they're referencing the same exact tag entity, but if we do in the future then we'd have to rewrite parts of the current design.

**Comment by martinjonsson01** [comment]
@JacobBredin 
> Small question: Why tags? or whats the problem that's being solved?

The issue was that you could not create a database table for `Contract`s without also having one for `Tag`s, since we've got multiple `Tag` instances associated with a single `Contract` (which can't be stored in a single row in the `Contract` table. 

So I had to refactor the tags to be an actual class instead of a string, and that was a bit more cumbersome than I initially thought since the input mechanism relied on them being strings. 😅

**Comment by JacobBredin** [comment]
> 

Ok, then I understand 👍Time to ship! 🚣

**Pull request merged**